src/python/zquantum/qcbm/ansatz.py

```diff
@@ -254,4 +259,63 @@ def get_number_of_parameters_by_layer(self) -> np.ndarray:
                 # A normal single qubit rotation layer
                 num_params_by_layer.append(self.number_of_qubits * 2)
 
-        return np.asarray(num_params_by_layer)
\ No newline at end of file
+        return np.asarray(num_params_by_layer)
+
+    def to_dict(self):
+        """Creates a dictionary representing a QCBM ansatz.
+
+        Returns:
+            dictionary (dict): the dictionary
+        """
+        dictionary = {
+            "schema": ANSATZ_SCHEMA,
+            "number_of_layers": self.number_of_layers,
+            "number_of_qubits": self.number_of_qubits,
+            "topology": self.topology,
+        }
+
+        return dictionary
+
+
+def save_qcbm_ansatz_set(qcbm_ansatz_set: List[QCBMAnsatz], filename: str) -> None:
+    """Save a set of qcbm_ansatz to a file.
+
+    Args:
+        qcbm_ansatz_set (list): a list ansatz to be saved
+        file (str): the name of the file
+    """
+    dictionary = {
+        "schema": ANSATZSET_SCHEMA,
+        "qcbm_ansatz": [ansatz.to_dict() for ansatz in qcbm_ansatz_set]
```

Comment:
I think the key here should be `qcbm_ansatz_set`

Reply:
Done!